src/python/ensembl/io/genomio/database/factory.py

```diff
@@ -140,7 +145,17 @@ def main() -> None:
         help="Enable BRC mode, i.e. use organism_abbrev for species, component for division",
     )
     parser.add_log_arguments()
-    args = parser.parse_args()
+    return parser.parse_args(arg_list)
+
+
+def main(arg_list: list[str] | None = None) -> None:
```

Comment:
why do we need arg_list?

Reply:
This was suggested by the page I was following to understand this approach. As far as I can tell, it mainly allows you for an easier unit testing of the `main()` method and reuse of the argument parsing declaration. Not my favourite implementation, but serves its purpose as PoC.